resources/views/errors/500.blade.php

```diff
@@ -4,6 +4,23 @@
 Internal Server Error
 @endsection
 
+@php
+/**
+    * Checks exception messages for POST/GET variables and removes them. Prevents dangerous things like environment variables
+    * from leaking while ensuring a meaningful and safe message can be displayed to the user (and reported to the dev team).
+*/
+function safeErrorMessage($exception) {
+    $safeErrorMessage = 'Unspecified error';
+    $exceptionMessage = $exception->getMessage();
+    if (preg_match('/(https?:\/\/.*[\?\&])/', $exceptionMessage)) {
+        $safeErrorMessage = preg_replace('/(https?:\/\/.*[\?\&][^\s]+)/', '---', $exceptionMessage);
+    } else {
```

Comment:
Would you mind providing some examples of what this will match? I'm worried about it not matching passwords, usernames, and hosts properly. I'm honestly a proponent of just removing the message altogether in favor of looking at the logs. That is the only way to completely prevent anything from being leaked.

Reply:
This regex matches the pattern `https://a.b&` to detect a GET or POST with parameters and it replaces the URI & parameters with `---`. The issue that we are having is secrets being exposed when the error message includes a GET or POST. This should eliminate the problem. I realize that there is a more robust method of checking the error message against the .env file, but I decided not to do this as it risks exposing the entire .env contents if my code does something unexpected. I agree that we could just remove the message, but it is helpful in troubleshooting... especially since I can't always troubleshoot an error immediately and errors often get reported well after they occur, making it challenging to track down in the logs. I'm open to your thoughts/comments on this.

Reply:
Gotcha that does make sense. I'll approve, but I wonder if there are any free but more robust logging solutions that we could use so we could just include the time the error occurred on that page and then we could just check the logs. I'll do some research